filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileDistributionRpcServer.java

```diff
@@ -74,22 +79,24 @@ private void declareFileDistributionMethods() {
     @SuppressWarnings({"UnusedDeclaration"})
     public final void getFile(Request req) {
         req.detach();
-        FileReference fileReference = new FileReference(req.parameters().get(0).asString());
-        log.log(LogLevel.DEBUG, "getFile() called for file reference '" + fileReference.value() + "'");
-        Optional<File> pathToFile = downloader.getFile(fileReference);
-        try {
-            if (pathToFile.isPresent()) {
-                req.returnValues().add(new StringValue(pathToFile.get().getAbsolutePath()));
-                log.log(LogLevel.DEBUG, "File reference '" + fileReference.value() + "' available at " + pathToFile.get());
-            } else {
-                log.log(LogLevel.INFO, "File reference '" + fileReference.value() + "' not found, returning error");
-                req.setError(fileReferenceDoesNotExists, "File reference '" + fileReference.value() + "' not found");
+        rpcDownloadExecutor.execute(() -> {
+            FileReference fileReference = new FileReference(req.parameters().get(0).asString());
```

Comment:
I prefer making it a private method and execute the private method. That gives more information on a stackdump. If not you have to consult code and to figure things out.

Reply:
Good point, I had not thought about that, will do